### PR TITLE
feat: [IOPLT-394] Bump Xcode version used to build the app

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR bumps xcode version on CI to be compliant with app store.
Uses arm runners to improve execution time.

## List of changes proposed in this pull request
- Changed the runner and setup on `release.yml` workflow

## How to test
Manually run a release workflow changing the workflow trigger or just check the previous run [here](https://github.com/pagopa/io-app/actions/runs/8362091374?pr=5611)

### CODE REVIEW

File: ios/fastlane/Fastfile [hunk 1]
```diff
@@ -244,7 +244,7 @@ platform :ios do
       merge_commit_filtering: "exclude_merges"
     )
 
-    # upload to App store
+    # # upload to App store
```
Comment (shadowsheep1) [inline ios/fastlane/Fastfile:247]: Double comment ;)

Comment (shadowsheep1) [inline ios/fastlane/Fastfile:247]:
```suggestion
    # upload to App store
```

**Review by LazyAfternoons — APPROVED** [review]
LGTM

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]
The best way to test this, is to put this workflow in a personal repo and with tamte luring inside the runner to see if all the commands run smoothly. What do you think?

**Comment by CrisTofani** [comment]
> The best way to test this, is to put this workflow in a personal repo and with tamte lurking inside the runner to see if all the commands run smoothly. What do you think?

@shadowsheep1 I ran an execution of the build process here https://github.com/pagopa/io-app/actions/runs/8362091374?pr=5611 just commenting the upload to app store job in the fastfile to avoid the rejection by TF

**Review by shadowsheep1 — APPROVED** [review]
LGTM

**Pull request merged**